builder_test.go

```diff
@@ -334,6 +334,273 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
+								},
+							}); s != "" {
+								t.Fatalf("Unexpected:\n%s\n", s)
+							}
+
+							expected := "Warning: redefining the following default process type with a process not marked as default: override-type"
+							h.AssertStringContains(t, stdout.String(), expected)
+
+							h.AssertEq(t, metadata.BuildpackDefaultProcessType, "")
+						})
+					})
+
+					when("there is a web process", func() {
+						when("buildpack API >= 0.6", func() {
+							it.Before(func() {
+								builder.Group.Group = []lifecycle.GroupBuildpack{
+									{ID: "A", Version: "v1", API: latestBuildpackAPI.String()},
+								}
+							})
+							it("shouldn't set it as a default process", func() {
+								bpA := testmock.NewMockBuildpack(mockCtrl)
+								buildpackStore.EXPECT().Lookup("A", "v1").Return(bpA, nil)
+								bpA.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+									Processes: []launch.Process{
+										{
+											Type:        "web",
+											Command:     "web-cmd",
+											Args:        []string{"web-arg"},
+											Direct:      false,
+											BuildpackID: "A",
+											Default:     false,
+										},
+									},
+								}, nil)
+
+								metadata, err := builder.Build()
+								if err != nil {
+									t.Fatalf("Unexpected error:\n%s\n", err)
+								}
+
+								if s := cmp.Diff(metadata.Processes, []launch.Process{
+									{
+										Type:        "web",
+										Command:     "web-cmd",
+										Args:        []string{"web-arg"},
+										Direct:      false,
+										BuildpackID: "A",
+										Default:     false,
+									},
+								}); s != "" {
+									t.Fatalf("Unexpected:\n%s\n", s)
+								}
+								h.AssertEq(t, metadata.BuildpackDefaultProcessType, "")
+							})
+						})
+
+						when("buildpack api < 0.6", func() {
+							it.Before(func() {
+								builder.Group.Group = []lifecycle.GroupBuildpack{
+									{ID: "A", Version: "v1", API: "0.5"},
+								}
+							})
+							it("should set it as a default process", func() {
+								bpA := testmock.NewMockBuildpack(mockCtrl)
+								buildpackStore.EXPECT().Lookup("A", "v1").Return(bpA, nil)
+								bpA.EXPECT().Build(gomock.Any(), config).Return(lifecycle.BuildResult{
+									Processes: []launch.Process{
+										{
+											Type:        "web",
+											Command:     "web-cmd",
+											Args:        []string{"web-arg"},
+											Direct:      false,
+											BuildpackID: "A",
+											Default:     false,
+										},
+										{
+											Type:        "not-web",
+											Command:     "not-web-cmd",
+											Args:        []string{"not-web-arg"},
+											Direct:      true,
+											BuildpackID: "A",
+											Default:     false,
+										},
+									},
+								}, nil)
+
+								metadata, err := builder.Build()
+								if err != nil {
+									t.Fatalf("Unexpected error:\n%s\n", err)
+								}
+
+								if s := cmp.Diff(metadata.Processes, []launch.Process{
```

Comment:
Maybe it's worth adding a comment here about why `Default = false` in the metadata.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/files#diff-5dce70063c4b1b07e611b3fcd32da0445a0b95970028c8992beb9c955e38320fR253.